sequence_processing_pipeline/ConvertJob.py

```diff
@@ -141,7 +142,49 @@ def run(self, callback=None):
                          changed.
         :return:
         """
-        job_info = self.submit_job(self.job_script_path, None, None,
-                                   exec_from=self.log_path, callback=callback)
+        try:
+            job_info = self.submit_job(self.job_script_path,
+                                       exec_from=self.log_path,
+                                       callback=callback)
+        except JobFailedError as e:
+            # When a job has failed, parse the logs generated by this specific
+            # job to return a more descriptive message to the user.
+            info = self.parse_logs()
+            # prepend just the message component of the Error.
+            info.insert(0, str(e))
+            raise JobFailedError('\n'.join(info))
```

Comment:
Is this necessary? Just confirming as it might duplicate the errors.

Reply:
By creating a subclass of PipelineError called JobFailedError, we allow all other PipelineErrors to pass right on through and be handled as they normally would while specifically handling the case where a Slurm job fails here. Specifically, we know that since we're in ConvertJob and it's a Slurm job that failed, we can reliably count on bcl-convert to generate a log in a specific location and be of a particular format.

The idea here is that we'll parse the log for all Errors and append them to the original JobFailed message string, giving it a much needed explanation. Then we'll simply re-raise the error with its extended description; since it's a child of PipelineError, it should be caught by the same code that is catching and displaying our other errors.